### ISSUE TITLE
Suboptimal performance

### ISSUE THREAD

**Issue by JulienMaufront** [teamtomo/fidder#14]
Hi Alister,

I tried Fidder but I could not tune the probability parameter so as to obtain a result as good the one obtained by Goldnet within WARP. Especially when we go for higher tilt images. (See the attached files : high/low tilts in which gold beads were erased by Goldnet or by Fidder). Also, the "optimal" values for both Goldnet/Fidder parameters are different (not equivalent ?) : "use score above 0.10" for Goldnet, and "--probability-threshold 0.005" for fidder.

![goldnet_low-tilt](https://user-images.githubusercontent.com/47780389/220136511-2a0e3496-bae8-4d54-8fe6-c92a2cbe0ac6.jpg)
![goldnet_high-tilt](https://user-images.githubusercontent.com/47780389/220136591-43b6d307-0ee6-45c5-a8c9-889a1d3b0cde.jpg)
![fidder_low-tilt](https://user-images.githubusercontent.com/47780389/220137470-5979c5c8-9ada-422b-867b-07ab255be870.png)
![fidder_high-tilt](https://user-images.githubusercontent.com/47780389/220137945-8ce307c6-68b7-4e6a-980d-2839ac678e60.png)

Is it something expected due to the quality of my dataset or does it reveals an issue while running fidder ?

When I run fidder i have some warnings : 

"--------------------------------------------------------------------------------------------------------------------------------------------------------------"
Lightning automatically upgraded your loaded checkpoint from v1.8.3.post1 to v1.9.2. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint --file ../../../../../home/daniel/.cache/fidder/fidder.ckpt`
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
Predicting: 0it [00:00, ?it/s]/home/daniel/.local/lib/python3.8/site-packages/pytorch_lightning/loops/epoch/prediction_epoch_loop.py:173: UserWarning: Lightning couldn't infer the indices fetched for your dataloader.
  warning_cache.warn("Lightning couldn't infer the indices fetched for your dataloader.")
Predicting DataLoader 0: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.12s/it]
/usr/local/lib/python3.8/dist-packages/scipy/interpolate/_fitpack2.py:1272: UserWarning: 
The coefficients of the spline returned have been computed as the
minimal norm least-squares solution of a (numerically) rank deficient
system (deficiency=34). If deficiency is large, the results may be
inaccurate. Deficiency may strongly depend on the value of eps.
  warnings.warn(message)
"--------------------------------------------------------------------------------------------------------------------------------------------------------------"

 Do I have to worry about the warnings and fix something ?

Thank you

**Comment by alisterburt** [comment]
Thank you for the detailed feedback @JulienMaufront ! It looks like performance is ~okay but definitely far from perfect - I think this is likely because your fiducials have quite variable size, something I don't have in my training set. 

Would you be able to send me ~10 motion corrected images by wetransfer or similar? I will retrain the model including this data and update the package, this should improve performance

> Do I have to worry about the warnings and fix something ?

The warnings both look okay but thanks for reporting them, I might silence them in an update :)

> Also, the "optimal" values for both Goldnet/Fidder parameters are different (not equivalent ?) : "use score above 0.10" for Goldnet, and "--probability-threshold 0.005"

Just FYI these values are not equivalent, in Warp that value only determines the threshold for picking particles - for the mask a hardcoded threshold of 0.10 is used. I hope you will be able to get away with higher thresholds once the model has seen some of your data :)

**Comment by alisterburt** [comment]
images at high tilt are probably the most useful but throw in some low tilt ones too for good measure

**Comment by JulienMaufront** [comment]
Thank a lot Alister !

You could find here some tilt images : https://we.tl/t-736A2hZBb4

Cheers

**Comment by alisterburt** [comment]
You're a star Julien, downloading now!

**Comment by alisterburt** [comment]
@JulienMaufront I've retrained including your data and uploaded the new data/weights - the new version of fidder should be deployed fairly soon, you can check progress here: https://github.com/teamtomo/fidder/actions/runs/4233472785

Your small fiducials basically become invisible at the pixel size the network works at (8A/px) but it should hopefully do a bit better than before - let me know how you get on!

**Comment by alisterburt** [comment]
v0.0.5 now available on PyPI - `pip install --upgrade fidder`

I'll close this as we've done what we can to improve things for you - thanks!

**Comment by alisterburt** [comment]
Ahh glad to hear it! I did use the values in the screenshots but it shouldn't cause too many issues, it's a small change :)

Glad you're finding it useful, thanks again for providing data!
